source/server/listener_impl.cc

```diff
@@ -526,31 +549,25 @@ void ListenerImpl::buildInternalListener() {
         config_.has_tcp_backlog_size() || config_.has_tcp_fast_open_queue_length() ||
         (config_.has_transparent() && config_.transparent().value())) {
       throw EnvoyException(
-          fmt::format("error adding listener '{}': has unsupported tcp listener feature",
-                      absl::StrJoin(addresses_, ",", Network::AddressStrFormatter())));
+          fmt::format("error adding listener '{}': has unsupported tcp listener feature", name_));
```

Comment:
The name isn't a good choice also, since 
https://github.com/envoyproxy/envoy/blob/a53743b20351288a37a61811f27bcda463dd32ad/source/server/listener_manager_impl.cc#L381-L383

And when we support multiple addresses which mix internal and IP address, log the addresses is better.

Reply:
Address is empty for internal listener I think in this branch.

Reply:
You append an internal addresses here https://github.com/envoyproxy/envoy/pull/22274/files#diff-b0d5e1a1317df8e22a556ee74d6f728379c5f85ce640a9404acc14bfe8192ad3R366-R367, so it shouldn't be empty

Reply:
Internal listeners are always deliberately named since that's the only way to refer to them in EnvoyInternalAddress. I think we can rework this if/when we add multiple address support for internal listeners.